docs/rfc/rfc-19-random-beacon-v2.adoc

```diff
@@ -0,0 +1,252 @@
+and 70 groups.
+
+Both on-chain and off-chain components proved to be successful but - just like
+with any other product - some pain points were identified by the users.
+The most problematic aspects were the construction of pricing scheme and too
+aggressive timeouts. This RFC aims at describing these problems and proposing
+changes that should make the beacon easier to operate and more consistent with
+tBTC v2.
+
+=== ETH-based pricing
+
+Having reimbursements for various on-chain actions in ETH was a nice feature but
+also complex in practice. Highly volatile gas prices on mainnet made the gas price
+ceiling value impossible to maintain up-to-date. Given that the gas price
+ceiling had to be higher than the current gas price and that the fee for future
+group creation was calculated based on the gas price ceiling, new groups were 
+getting created too often. 
+
+With too many groups being created, operators were spending too much ETH on
+ticket submission and too low rewards were getting accumulated per group making
+the reward withdrawal operation not profitable enough or not profitable at all
+in the case of small stakers.
+
+Extra submitter rewards added on top of reimbursements were hard to understand
+for staking providers and calculating the provider’s fee was complicated given
+the non-reimbursable cost of ticket submission. Surprisingly, it is much easier
+for staking providers to calculate their fee and costs in tBTC v1, having a
+clean rule that no operations are reimbursed and the only reward they receive
+is KEEP plus TBTC fees.
+
+=== Aggressive timeouts
+
+Relay entry timeouts were pretty aggressive. Set to 64x6 = 384 blocks, they were
+giving about 1.5 hours to submit a relay entry before the minimum stake from each
+group member was slashed. This timeout proved to be not enough to notify,
+investigate, and fix a problem. Three groups were slashed in a row during the
+November 11th incident that begun around 08:00 UTC, before any US-based
+operators noticed the problem.
+
+
+== Proposal
+
+=== Group creation
+
+New groups should be created with a fixed, governable frequency of relay
+requests. Instead of a ticket-based approach for group selection, we should
+use sortition pools. Group creation start transaction should be embedded into
+relay request transaction and should lock a sortition pool. From this moment,
+no operator can enter or leave the pool. Once a new relay entry appears on the
+chain, all off-chain clients should perform group selection optimistically, 
+using the new entry value and a view sortition pool function call.
+After determining group members, clients should perform DKG as usual. 
+Once the group submits the result to the chain, a challenge period starts.
+
+During the challenge period, anyone can notify that the submitted DKG result
+contains group members not selected by the pool. This leads to slashing all
+members who signed the result and the notifier receives 5% from the slashed amount.
```

Comment:
What should happen (who will get paid the 5%) in case there will be more than one notification (is that even possible)?